secboot/secboot_tpm.go

```diff
@@ -436,7 +440,125 @@ func runFDERevealKeyCommand(stdin []byte) (output []byte, err error) {
 	return nil, fmt.Errorf("internal error: systemd-run did not honor RuntimeMax=%s setting", fdeRevealKeyRuntimeMax)
 }
 
+var noSecbootV2FileErr = errors.New("no secboot v2 file")
+
+// We have to deal with the following cases:
+// 1. Key created with v1 data-format on disk (raw encrypted key), v1 hook reads the data
+// 2. Key created with v2 data-format on disk (json), v1 hook created the data (no handle) and reads the data (hook output not json but raw binary data)
+// 3. Key created with v1 data-format on disk (raw), v2 hook
+// 4. Key created with v2 data-format on disk (json), v2 hook [easy]
 func unlockVolumeUsingSealedKeyFDERevealKey(name, sealedEncryptionKeyFile, sourceDevice, targetDevice, mapperName string, opts *UnlockVolumeUsingSealedKeyOptions) (UnlockResult, error) {
+	// Try the v2 format first and fallback to v1 if needed.
+	res, err := unlockVolumeUsingSealedKeyFDERevealKeyV2(name, sealedEncryptionKeyFile, sourceDevice, targetDevice, mapperName, opts)
+	if err == noSecbootV2FileErr {
+		// try the "old" keyfile (which is plain binary)
+		return unlockVolumeUsingSealedKeyFDERevealKeyV1(name, sealedEncryptionKeyFile, sourceDevice, targetDevice, mapperName, opts)
+	}
+	return res, err
+}
+
+func unlockVolumeUsingSealedKeyFDERevealKeyV2(name, sealedEncryptionKeyFile, sourceDevice, targetDevice, mapperName string, opts *UnlockVolumeUsingSealedKeyOptions) (UnlockResult, error) {
+	res := UnlockResult{IsEncrypted: true, PartDevice: sourceDevice}
+
+	// XXX: fugly, we need a way to pass the key-name to the hook for
```

Comment:
Just discussed this with Chris and Samuele and the agreement is to sent "deprecated-$random-12-chars" so that we don't break the existing "avnet" project which uses this only for the unseal case (but does not cares about the actual value).

Reply:
This is done now